### PR TITLE
Remove pylint

### PR DESCRIPTION
**Pull request by potiuk**
We've agreed during the voting process that Pylint support
should be disabled: https://lists.apache.org/thread.html/r9e2cc385db8737ec0874ad09872081bd083593ee29e8303e58d21efb%40%3Cdev.airflow.apache.org%3E

This PR:

* removes all # pylint comments
* removes pylint pre-commits and related scripts/files
* removes CI jobs running pylint checks
* removes documentation about pylint
* removes unnecessary #noga (adds pre-commit for that)
* fixes some remaining `pydocstyle` errors after removing #noqa's

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Anticipating the result of voting - I prepared this PR that removes pylint

**Review by mik-laj — APPROVED** [review]

File: airflow/api/client/json_client.py [hunk 1]
```diff
@@ -31,12 +31,12 @@ def _request(self, url, method='GET', json=None):
...
         if json is not None:
             params['json'] = json
-        resp = getattr(self._session, method.lower())(**params)  # pylint: disable=not-callable
+        resp = getattr(self._session, method.lower())(**params)
         if not resp.ok:
             # It is justified here because there might be many resp types.
             try:
                 data = resp.json()
-            except Exception:  # noqa pylint: disable=broad-except
+            except Exception:  # noqa
```
Comment (mik-laj) [inline airflow/api/client/json_client.py]: I suspect that this `noqa` tag has no effect on any tool now.  We should use [`yesqa`](https://github.com/asottile/yesqa) to delete the unused tag, but we can do it as a separate PR.

Comment (potiuk) [reply airflow/api/client/json_client.py]: I think flake also complains about this case. But yeah. yesqa is a good idea. I will run it now no problem.

Comment (mik-laj) [reply airflow/api/client/json_client.py]: Flake dont support this case. In other place, we have broad exception without `noqa` tag.

**Comment by uranusjr** [comment]
We should add a `.git-blame-ignore-revs` file for this commit.

**Comment by uranusjr** [comment]
(The changeset is so big GitHub is having trouble rendering, so I’m going to assume this is correct…)

**Comment by potiuk** [comment]
I deed. Might be a good point also for few other changes like that. But this has to be a separate PR after we know the  commit hash :)

**Comment by potiuk** [comment]
I've added `yesqa` to pre-commits and removed all the unnecessary #noqa's. It removed a bit too many of those initially. It does not understand some pydocstyle problems but I figured it will be easier to remove a "public method no doc" error `D103 | Missing docstring in public function`  (which I think is far too much to expect) and fixed remaining small docstyle errors which were earlier ignored by #noqa.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -644,12 +642,6 @@ repos:
         files: ^airflow/www/static/js/
         entry: scripts/ci/static_checks/www_lint.sh
         pass_filenames: false
-      - id: bats-in-container-tests
-        name: Run in container bats tests
-        language: system
-        entry: ./scripts/ci/pre_commit/pre_commit_in_container_bats_test.sh
-        files: ^tests/bats/in_container/.*\.bats$|^scripts/in_container/.*sh
-        pass_filenames: false
```
Comment (ashb) [inline .pre-commit-config.yaml:652]: Seems unrelated to pylint?

Comment (ashb) [reply .pre-commit-config.yaml:652]: Oh it moved.

File: airflow/_vendor/connexion/decorators/parameter.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import inflection
 
 from ..http_facts import FORM_CONTENT_TYPES
-from ..lifecycle import ConnexionRequest  # NOQA
+from ..lifecycle import ConnexionRequest
```
Comment (ashb) [inline airflow/_vendor/connexion/decorators/parameter.py]: Generally I'd say we shouldn't check/change code under `airflow/_vendor/` -- so we should exclude this from the Yesqa check if it's complaining and not otherwise touch these files

**Review by ashb — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (potiuk) [reply .pre-commit-config.yaml:652]: It's moved to after the `build` step - the check failed if the build has not been run before (discovered it while testing this change). I can separate it out, but since we are changing the pre-commit anyway quite heavily, fixing it in the "big" pylint cleanup PR might be a good idea.

File: airflow/_vendor/connexion/decorators/parameter.py (hunk 3 above)
Comment (potiuk) [reply airflow/_vendor/connexion/decorators/parameter.py]: Good point. fixing it

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ashb) [reply .pre-commit-config.yaml:652]: Nah, it's good.

**Review by ashb — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
Booom
